### ISSUE TITLE
missing x.cov

### ISSUE THREAD

**Issue by ecmerkle** [yrosseel/lavaan#169]
The line referenced below is causing one or two blavaan tests to fail because x.cov is NULL. I haven't isolated the problem (have not looked hard), but I can remember having previous trouble with the difference between missing="ml" and missing="ml.x". I guess this might be related? (fwiw, I had been trying to stick with missing="ml" because "ml.x" was making my life difficult in various ways)

I am not expecting any changes in lavaan, but any extra info would be helpful for me to solve the problem in blavaan. Thanks...

https://github.com/yrosseel/lavaan/blob/4ecede812aef78d945981ed2873ba960e9c9388a/R/lav_mvnorm_missing.R#L137

**Comment by ecmerkle** [comment]
Just some follow up on this issue: I have tracked the problem to models with fixed.x=TRUE and missing="ml". For example

```
model <- ' y1 ~ x1 + x2 '
fit <- sem(model, data=PoliticalDemocracy, fixed.x=TRUE, missing="ml")
```

i will keep digging, but I am happy to find that the error can be triggered outside of blavaan!

**Comment by yrosseel** [comment]
This was part of an (unfinished) transition to always subtract the loglikelihood of the (fixed) 'x' variables, when the model loglikelihood is computed. This was already the case for complete data, but not for incomplete data. Now fixed in lavaan 0.6-6.1516. (Still not implemented for the multilevel case).
